[wujiaweijoanna] iP

wujiaweijoanna opened the pull request:
# Alice
> "It’s no use going back to yesterday, because I was a different person then." - Lewis Carroll [source](https://toppsta.com/blog/view/alice-in-wonderland-quotes)

Alice frees your mind of having to remember things you need to do. It's,

+ text-based
+ easy to use
+ ~~FAST~~ SUPER _FAST_ to use

All you need to do is,

1. download it from [here](https://github.com/wujiaweijoanna/ip).
2. add your tasks.
3. let it manage your tasks for you 👍 

And it is **FREE**!

Features:

- [x] Managing tasks
- [ ] Reminders (coming soon)

---

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:
``` java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}

```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

wujiaweijoanna committed:
Rename, greet, exit

wujiaweijoanna committed:
Echo commands

wujiaweijoanna committed:
Add, list tasks

wujiaweijoanna committed:
Mark task as done

wujiaweijoanna committed:
Todo, Event, Deadline

wujiaweijoanna committed:
Automated text UI testing

wujiaweijoanna committed:
Handle errors

wujiaweijoanna committed:
Delete tasks

wujiaweijoanna committed:
Use java enums

wujiaweijoanna committed:
Save the tasks

wujiaweijoanna committed:
Merge branch 'branch-Level-7'

wujiaweijoanna committed:
Change time format

wujiaweijoanna committed:
Merge branch 'branch-Level-8'

wujiaweijoanna committed:
Extract out closely related code as classes

wujiaweijoanna committed:
Merge branch 'branch-A-MoreOOP'

wujiaweijoanna committed:
Organize the classes into one package

wujiaweijoanna committed:
Merge branch 'branch-A-Packages'

wujiaweijoanna committed:
Merge branch 'add-gradle-support'

wujiaweijoanna committed:
Set up gradle

wujiaweijoanna committed:
Add JUnit tests

wujiaweijoanna committed:
Merge branch 'branch-A-JUnit'

wujiaweijoanna committed:
Add JavaDoc comments

wujiaweijoanna committed:
Comply with the coding standard

wujiaweijoanna committed:
Find a task by searching for a keyword

wujiaweijoanna committed:
Merge branch 'branch-A-JavaDoc'

wujiaweijoanna committed:
Merge branch 'branch-A-CodingStandard'

wujiaweijoanna committed:
Merge branch 'branch-Level-9'

wujiaweijoanna committed:
Use CheckStyle to detect coding style violations

wujiaweijoanna committed:
Merge branch 'branch-A-CheckStyle'

bbbbcccd commented on src/main/java/alice/Parser.java:
```diff
@@ -0,0 +1,298 @@
+import java.time.format.DateTimeFormatter;
+
+/**
+ * Parses user commands and performs corresponding actions.
```
Clear description of the responsibility of the class 👍

bbbbcccd commented on src/main/java/alice/Parser.java:
```diff
@@ -0,0 +1,298 @@
+        switch (input) {
+        case "bye":
+            return CommandType.BYE;
+            // Fallthrough
```
I noticed a few of these "Fallthrough" comments. What are they indicating?

bbbbcccd commented on src/main/java/alice/Parser.java:
```diff
@@ -0,0 +1,298 @@
+     *
+     * @param input The user input string containing the "find" command followed by the keyword
+     *              to search for in the task list.
+     *              The input format should be "find [keyword]".
```
Nice formatting of the JavaDoc! 👍

bbbbcccd reviewed (commented):
Lgtm! Very clean code that follows the coding standard 🥇

wujiaweijoanna committed:
Implement GUI using JavaFX

wujiaweijoanna committed:
Merge branch 'branch-Level-10'

JellyPenguinnn commented on src/main/java/alice/utility/TaskList.java:
```diff
@@ -0,0 +1,139 @@
+     * @param keyword The keyword to search for in task descriptions.
+     * @return A TaskList containing tasks that match the keyword.
+     */
+    public TaskList findTask(String keyword) {
```
This method could be renamed to findTaskByKeyword to be more specific

JellyPenguinnn commented on src/main/java/alice/utility/TaskList.java:
```diff
@@ -0,0 +1,139 @@
+        return this.list;
+    }
+
+    public int getSize() {
```
Can use getTaskCount to make it clearer that is method is to return the number of tasks

JellyPenguinnn commented on src/main/java/alice/utility/TaskList.java:
```diff
@@ -0,0 +1,139 @@
+     * @return A TaskList containing tasks that match the keyword.
+     */
+    public TaskList findTask(String keyword) {
+        ArrayList<Task> temp = new ArrayList<>();
```
Can use a more descriptive name such as matchingTasks

JellyPenguinnn reviewed (commented):
The code is well-structured with a logical grouping of related functionality.
It is generally easy to read and follow, with clear methods and class names that align well with their purpose.
Btw the cats are cute ;)
Well done !!!

wujiaweijoanna committed:
Use assertions

The Parser class has various methods that depend on certain assumptions.

There is a lack of enforcement of the assumptions.

Assertion is added to enforce these assumptions.

Assertions are effective in validating assumptions and helping
developers identify issues early in the development process.

wujiaweijoanna committed:
Improve code quality

The TaskList class has method names that could be more descriptive.

Unclear names make the code difficult to read.

Several methods and variables are renamed.

Using descriptive names for methods and variables makes the codebase
easier to understand and navigate.

wujiaweijoanna committed:
Merge pull request #1 from wujiaweijoanna/branch-A-Assertions

Add assertions

wujiaweijoanna committed:
Merge branch 'master' into branch-A-CodeQuality

wujiaweijoanna committed:
Merge pull request #2 from wujiaweijoanna/branch-A-CodeQuality

Improve code quality

wujiaweijoanna committed:
Set up CI

wujiaweijoanna committed:
Recognize and deal with duplicate tasks

wujiaweijoanna committed:
Merge branch 'branch-BCD-Extension'

wujiaweijoanna committed:
Update README

wujiaweijoanna committed:
Handle find command exception
